srml/assets/src/lib.rs

```diff
@@ -95,11 +97,13 @@ decl_module! {
 		/// Destroy any assets of `id` owned by `origin`.
 		fn destroy(origin, id: AssetId) -> Result {
 			let origin = ensure_signed(origin)?;
+			let origin_account = (id, origin.clone());
```

Comment:
this alteration results in meandering, less concise code, more bloat and slower execution. these two lines can be dropped and the following two lines be switched, just as it was before.

Reply:
It's not an issue because:
- It can only be done "for free" once per zero-containing account, which implies a price has already been paid for the creation and subsequent transfer emptying that account;
- A base fee is already paid for that transaction to be included at all so it is not a zero-cost attack in any sense anyway.